installer.sh

```diff
@@ -0,0 +1,2307 @@
+            summary+="✗ KVM: Libvirt not running\n"
+            status_ok=false
+        fi
+    fi
+
+    # 6. Usage Server Validation
+    if [[ " ${SELECTED_COMPONENTS[@]} " =~ " usage " ]]; then
+        if systemctl is-active --quiet cloudstack-usage; then
+            summary+="✓ Usage Server: Running\n"
+        else
+            summary+="✗ Usage Server: Not running\n"
+            status_ok=true   # not critical for zone deployment
+        fi
+    fi
+
+    # Log the result
+    echo -e "Validation Summary:\n$summary" >> "$CS_LOGFILE"
+
+    # Display final result in dialog
+    if $status_ok; then
+        dialog --backtitle "$SCRIPT_NAME" \
+               --title "Validation Summary" \
+               --colors \
+               --msgbox "\Z2✓ All components are properly configured!\n\n\ZnComponent Status:\n\n$summary" 20 70
+        return 0
+    else
+        dialog --backtitle "$SCRIPT_NAME" \
+               --title "Validation Summary" \
+               --colors \
+               --msgbox "\Z1⚠ Some components need attention!\n\n\ZnComponent Status:\n\n$summary" 20 70
+        return 1
+    fi
+}
+
+configure_prerequisites() {
+    setup_network
+    setup_repo
+    update_system
+}
+
+validate_setup_pre_req() {
+    check_root
+    check_kvm_support
+    check_system_resources
+    detect_os
+    install_base_dependencies
+}
+
+custom_install() {
+    select_components
+    install_configure_components
+}
+
+all_in_one_box() {
+    PROMPT=0
+    if dialog --backtitle "$SCRIPT_NAME" \
+           --title "All-in-One Installation" \
+           --yesno "You have selected all components for installation. This will configure a complete CloudStack setup on this single machine.\n\nProceed with All-in-One installation?" 12 60; then
+        SELECTED_COMPONENTS=("nfs" "management" "agent" "usage")
+        install_configure_components
+    else
+        dialog --backtitle "$SCRIPT_NAME" \
+               --title "All-in-One Installation" \
+               --msgbox "Installation cancelled by user." 6 60
+        exit 0
+    fi
+}
+
+main() {
+    log "CloudStack Installer Script Started"
+    load_tracker
+    validate_setup_pre_req
+    configure_prerequisites
+    local temp_file=$(mktemp)
+    dialog --backtitle "$SCRIPT_NAME" \
+           --title "Installation Options" \
+           --menu "Select an option:" 12 50 4 \
+           1 "All-in-One Installation" \
+           2 "Custom Installation" \
+           3 "Configure CloudStack Repository" \
+           4 "Deploy CloudStack Zone" 2> "$temp_file"
+    if [[ $? -ne 0 ]]; then
+        error_exit "Installation option selection cancelled by user"
+    fi
+    local option=$(< "$temp_file")
+    case $option in
+        1)
+            all_in_one_box
+            ;;
+        2)
+            custom_install
+            ;;
+        3)
+            setup_repo
+            ;;
+        4)
+            setup_network
+            ;;
+        5)
+            deploy_zone
+            ;;
```

Comment:
dialog options and their handling doesn't match?

Also, maybe consistent naming:
- all_in_one_install
- custom_install
- configure_repo
- configure_network (not seeing this in dialog options though)
- deploy_zone